crates/solvers/src/boundary/liquidity/weighted_product.rs

```diff
@@ -38,7 +38,7 @@ pub fn to_boundary_pool(address: H160, pool: &liquidity::weighted_product::Pool)
                 },
             ))
         })
-        .collect::<Option<HashMap<_, _>>>()?;
+        .collect::<Option<BTreeMap<_, _>>>()?;
```

Comment:
The compiler can deduce the type which means you don't have to import `BTreeMap`.
```suggestion
        .collect::<Option<_>>()?;
```

Reply:
7d31e635d98917da4d35aeea68dabc42df11e20a